core/src/main/codegen/templates/Parser.jj

```diff
@@ -7313,6 +7361,12 @@ SqlNode JdbcFunctionCall() :
         s = span();
     }
     (
+        LOOKAHEAD(1)
+        call = DateDiffFunctionCall() {
+            name = call.getOperator().getName();
+            args = new SqlNodeList(call.getOperandList(), getPos());
+        }
```

Comment:
why have you added a call to `DateDiffFunctionCall` but not `DatetimeDiffFunctionCall`?

with this omission, I don't understand how the tests could pass.

Reply:
I mistakenly created DatetimeDiffFunctionCall as a SqlNode instead of a SqlCall so it was circumventing this. I'm not entirely sure how it was passing, but I've updated it to be consistent with the other function calls.